### DISCUSSION TITLE
Data Test Eror Di Scale

### DISCUSSION THREAD

**Discussion by LelaNovi** [teamalgoritma/community#802] [Q&A] [answered]
<img width="1680" alt="image" src="https://github.com/teamalgoritma/community/assets/145350763/6fa610c4-49fe-4812-be2c-06c292baca3a">
<img width="1680" alt="Screenshot 2023-11-20 at 21 44 13" src="https://github.com/teamalgoritma/community/assets/145350763/42663cf6-52fe-4748-82cd-8769e31f5c19">

hasil prediksi masih belum bisa hijau di leaderboard,apakah scaling di data test bermasalah??

**Comment by kinantytasyao** [comment, marked as answer]
Dear @LelaNovi 

Error tesebut menunjukkan bahwa terjadi ketidaksesuaian antara test dengan scale centernya. Meninjau code yan diberikan Bu Lela silakan untuk menghapus code yan mengubah data train hasil scale() menjadi `dataframe` karena seharusnya data hasil scale adalah matrix array dan ketika di test.csv tidak perlu menghapus kolom "strength" agar kolom-kolom di test sama persis dengan data train yang digunakan untuk pemodelan

![image](https://github.com/teamalgoritma/community/assets/79038260/0b9f650c-a765-4c11-984b-30530e331246)
![image](https://github.com/teamalgoritma/community/assets/79038260/772e9687-2c8d-4587-8b6e-485c14a85193)

**Comment by LelaNovi** [comment]
sudah dicoba dengan menscale fungsi ini MAE yang di dapat 11,9
<img width="1680" alt="Screenshot 2023-11-22 at 06 32 10" src="https://github.com/teamalgoritma/community/assets/145350763/ab1ce99f-1625-462e-903b-4193b55fe357">


saya coba dengan model baru terlampir dibawah 
<img width="1680" alt="Screenshot 2023-11-22 at 06 35 44" src="https://github.com/teamalgoritma/community/assets/145350763/25bcc361-80fa-4627-9f47-5a0bf099cbd8">

dicoba dengan data baru tidak menunjukkan hasil submissionyang memuaskan R square 88% MAE 13


apakah yang di data baru perlu scale lagi  dan dibuang outlier bagaimana menskalanya ?

**Reply by Yusraf002** [reply]
Dear @LelaNovi 

Disini apabila Ibu ingin melakukan _tunning_ model dengan cara menghilangkan _outlier_nya, Ibu bisa melakukannya dengan tahapan berikut ini:

1. mendeteksi outlier dari setiap variabel menggunakan fungsi `boxplot.stats(data$kolom)$out`, dimana fungsi tersebut akan mengeluarkan nilai outlier dari kolom tertentu
2. lakukan filter dengan mengeluarkan nilai outlier dari kolom tersebut
3. ulangi kedua tahapan diatas untuk setiap variabel yang memiliki outlier

untuk contoh penggunaannya ibu bisa coba ikuti tahapan pada sumber bacaan berikut: https://statsandr.com/blog/outliers-detection-in-r/

setelah keseluruhan data dirasa sudah tidak lagi memiliki _outlier_, ibu bisa lakukan tahapan _train & test spliting_ dan dilanjutkan dengan tahapan scaling seperti yang sudah ibu lakukan sebelumnya.

semoga jawaban ini membantu. 

Best Regards, 
Team Algoritma

**Comment by kinantytasyao** [comment]
Halo @LelaNovi  , karena tidak ada konfrimasi lebih lanjut dan untuk menjaga agar forum diskusi tetap berjalan kondusif, kami izin melakukan lock/close diskusi ini. Semoga dapat dipahami.
